lasagne/layers/recurrent.py

```
@@ -99,6 +105,11 @@ def __init__(self, incoming, input_to_hidden, hidden_to_hidden,
         self.gradient_steps = gradient_steps
         self.grad_clipping = grad_clipping
         self.unroll_scan = unroll_scan
+        self.precompute_input = precompute_input
+
+        if precompute_input and gradient_steps != -1:
```

We should probably also raise that error/similar error if scan_uroll is true.